### PR TITLE
Created node package to manage installation of javascript schema validator

### PR DESCRIPTION
**Pull request by rfultz**
#34 

There were a couple of bundle size management issues in the way this ticket was originally written. The ticket was updated to implement code tree shaking and Angular lazy load strategies to more efficiently manage the download of the large number of JSON schema definition files and their bundle size.

### CODE REVIEW

**Comment by mjtravers** [comment]
The SonarCloud Code Analysis is reporting 0% code coverage for the new javascript code. This is inaccurate as there are unit tests for this code. The issue is that CircleCI is configured to send code coverage reports to SonarCloud for the javascript code as it is for the python code. This ticket has been generated to fix this issue: #60

File: fecfile_validate_js/tests/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+    const result = validate(f3xSchema, thisData);
+    assertEquals(result, []);
+  },
+});
+
+Deno.test({
+  name: "filer_committee_id_number should not be ''",
+  fn: () => {
+    const thisData = { ...perfectForm_F3X, ...{ filer_committee_id_number: '' } };
+    // TODO: the next line is the goal. Ajv sees '' as a value and we want it to be evaluated as null
```
Comment (toddlees) [inline fecfile_validate_js/tests/index.test.ts:63]: why is this the goal? `''` is different from null.  If we configure our schema such that a field is required and the min length is 0, we have approved `''` as a valid entry. I don't believe that we _should_ be allowing `''` but I think that's a matter of setting the `minLength` and or the `pattern` properly.  My reasoning for that is that length is a string property and not a matter of defined or undefined.  `minLenght` and `pattern` are ways that we define string rules.  `required` is how is define whether or not a field must be defined.

File: schema/F3X.json [hunk 2]
```diff
@@ -1,2560 +1,2158 @@
...
+                "FIELD_FORM_ASSOCIATION": null
+            }
+        },
+        "state": {
+            "title": "STATE",
+            "description": "",
+            "type": ["string", "null"],
+            "minLength": 2,
+            "maxLength": 2,
+            "pattern": "^[A-Z]{2}$",
```
Comment (toddlees) [inline schema/F3X.json:162]: This is a good improvement on the pattern!

File: schema/F3X.json [hunk 3]
```diff
@@ -1,2560 +1,2158 @@
...
+                "FIELD_FORM_ASSOCIATION": null
+            }
+        },
+        "filer_committee_id_number": {
+            "title": "FILER COMMITTEE ID NUMBER",
+            "description": "",
+            "type": "string",
+            "minLength": 9,
+            "maxLength": 9,
+            "pattern": "^[C|P][0-9]{8}$|^[H|S][0-9]{1}[A-Z]{2}[0-9]{5}$",
```
Comment (toddlees) [inline schema/F3X.json:52]: Super useful definition of committee id

**Review by toddlees — CHANGES_REQUESTED** [review]

**Review by toddlees — APPROVED** [review]
Approving the package

**Pull request merged**